src/kernels/gpu_reference_kernel/naive_conv.cpp

```diff
@@ -1025,6 +1036,132 @@ inline __device__ void naive_conv_fwd_ndhwc(const src_data_t* __restrict__ p_in,
+                                            Strides5D in_strides,
+                                            Strides6D wei_strides,
+                                            Strides5D out_strides,
+                                            int di,
+                                            int hi,
+                                            int wi,
+                                            int n,
+                                            int k_per_group,
+                                            int c_per_group,
+                                            int do_,
+                                            int ho,
+                                            int wo,
+                                            int sz,
+                                            int sy,
+                                            int sx,
+                                            int dz,
+                                            int dy,
+                                            int dx,
+                                            int pz,
+                                            int py,
+                                            int px,
+                                            int fz,
+                                            int fy,
+                                            int fx,
+                                            int group)
+{
+    /*
+     *  need to compute total output pixel: `group * n * do_ * ho * wo *
+     * k_per_group`.
+     *  to distribute this workload, let one workgroup compute `ho * wo *
+     * k_per_group` pixel,
+     *  hence need `group * n * do_` workgroups (grid_size).
+     */
+    int k             = k_per_group * group;
+    int c             = c_per_group * group;
+    int thread_length = ho * wo * k_per_group;
+    int bid           = blockIdx.x;
+    int ido           = bid % do_;
+    int in            = (bid / do_) % n;
+    int ig            = bid / (n * do_);
+
+    //  TODO: what to do with ig * c_per-group and similarly ig * k_per_group
+    // p_in += static_cast<size_t>(in) * di * hi * wi * c + static_cast<size_t>(ig) * c_per_group;
+    // p_wei += static_cast<size_t>(ig) * k_per_group * fz * fy * fx * c_per_group;
+
+    p_in += static_cast<size_t>(in) * di * hi * wi * c + static_cast<size_t>(ig) * c_per_group;
+    // assumes that group G is the highest dimension in the layout
+    p_wei += static_cast<size_t>(ig) * wei_strides[5];
+
+    // p_out += static_cast<size_t>(in) * do_ * ho * wo * k + static_cast<size_t>(ido) * ho * wo * k +
+             // static_cast<size_t>(ig) * k_per_group;
+    p_out += static_cast<size_t>(in) * out_strides[4] + static_cast<size_t>(ido) * out_strides[3]
+      + ??;
+
+
+    for(int tid = threadIdx.x; tid < thread_length; tid += blockDim.x)
+    {
+        int ik  = tid % k_per_group;
+        int iwo = (tid / k_per_group) % wo;
+        int iho = tid / (k_per_group * wo);
+
+        double value = .0f;
```

Comment:
Either it should be `float value = .0f;` or `double value = 0.0;` or even better - `auto value = static_cast<acc_data_t>(0);`

Reply:
fixed in latest state.